lib/Sema/CSBindings.cpp

```diff
@@ -1631,8 +1640,22 @@ void BindingSet::dump(llvm::raw_ostream &out, unsigned indent) const {
     attributes.push_back("subtype_of_existential");
   auto literalKind = getLiteralKind();
   if (literalKind != LiteralBindingKind::None) {
-    auto literalAttrStr = ("[literal: " + getLiteralBindingKind(literalKind)
-                        + "]").str();
+    std::string literalAttrStr;
+    if (literalKind == LiteralBindingKind::Atom) {
+      if (TypeVar->getImpl().getAtomicLiteralKind().hasValue()) {
+        auto atomKind = TypeVar->getImpl().getAtomicLiteralKind().getValue();
```

Comment:
```suggestion
      if (auto atomKind = TypeVar->getImpl().getAtomicLiteralKind()) {
```

Reply:
Here, `TypeVar->getImpl().getAtomicLiteralKind()` returns an `llvm::Optional<swift::ExprKind>`, so I'm checking if it `hasValue()` before `getValue()`. I could rewrite it like below, if that's easier to read? 
```
if (auto &impl = TypeVar->getImpl(); impl.getAtomicLiteralKind().hasValue()) {
   auto atomKind = impl.getAtomicLiteralKind().getValue();
   auto atomLiteral = impl.getAtomLiteral(atomKind).str();
   ...
}
```